src/tests/system/tests/test_ipa.py

```diff
@@ -220,6 +220,80 @@ def test_ipa__hostpublickeys_with_non_default_port(client: Client, ipa: IPA, pub
         )
 
 
+@pytest.mark.importance("medium")
+@pytest.mark.integration
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__user_authorized_public_ssh_key(client: Client, ipa: IPA):
+    """
+    :title: Check sss_ssh_authorizedkeys succeeds
+    :setup:
+        1. Create users 'user1' and 'user2'
+        2. Configure and start SSSD with SSH responder
+        3. Lookup 'user1' and create SSH key pair
+        4. Add public key to IPA user 'user1'
+    :steps:
+        1. Lookup 'user1' and 'user2' using sss_ssh_authorizedkeys
+    :expectedresults:
+        1. Lookup for 'user1' passes and 'user2' fails
+    :customerscenario: False
+    """
+    client.sshd.config_set(
+        [{"AuthorizedKeysCommand": "/usr/bin/sss_ssh_authorizedkeys", "AuthorizedKeysCommandUser": "nobody"}]
+    )
+    client.sshd.reload()
+
+    user = ipa.user("user1").add()
+    client.sssd.enable_responder("ssh")
+    client.sssd.start()
+
+    result = client.tools.getent.passwd("user1")
+    assert result is not None
+    assert result.name is not None
+    assert result.home is not None
+
+    keys = client.tools.sshkey.generate(result.name, result.home)
+    user.modify(sshpubkey=keys[0])
+
+    # sss_ssh_authorizedkeys command does not contain any output if a key is found
```

Comment:
Hi,

I think this comment is wrong, if a key is found the
key is returned and I think it should be checked in the test is the
returned key is the expected one.

For the negative test, it looks like 'user2' is not created and hence
the error return code is expected because the user is missing. If there
is not ssh key assigned I would expect that sss_ssh_authorized_keys
returns '0' and just no key.

bye,
Sumit

Reply:
So the test isn't working correctly. Looking into further, it looks like the public key is not the same after adding it to IPA, maybe an encoding thing?

```
print("IPA_KEY::", user.get(attrs=["ipaSshPubKey"]))

IPA_KEY:: {'ipaSshPubKey':['c3NoLXJzYSBBQUFBQjNOemFDMXljMkVBQUFBREFRQUJBQUFCZ1FDK2x1MGZYZ3dYR3VWY096NDFiT0tQdG44T21uMDVlQ21UTXIzQzZDOUNZUVdZUVlSS2xDdlRxcTVpN2xqbWNXRUZrL2RrTlFZZkwycFdMMnE1ejl3TzJKSkhMQTFicERIdTJFSk9henF1Y2FXV3Q0bk9BZFNNRkNLYjdEbGUva0RZODRxSE40Z0ZEVUdpR1h3NHpOTzN1c1hHcWRyM2NzeDlPa2JYSTJsQUNhdU5tMVFNZzZFYnBtK3ZpajlZVXRmMUJPS3V3OGtHY3VTaEllZXJXdXRoNURJalZrM2pBYk5tSWRaNTFOd1ZndjhRcUR6SjVZLzZIdjZrR2FWelpyS3ZZSTJMcXNHdFdEQ0RUby8yc0VYTDU2WU9LOXhKVktvZy91NC9iU2dya29XWDRCK2FjTThtT1NCQmtMQ2RPaGxlbHYzYjRuT0hCOHZ1QWMwbEd3dUFhckR6aTFJRWxEdHdQNVBZMUZ1RDVoZkN6QkZ1bCt3ckdPOElnV3crWjlhWU9YRHRFY0xreG5kc3pQU25jTlFBTWIrM0FnVUhXbm13MSt4dE16MjdGUXFxamRHOWlhTExuZk1zK3E3c0hXTmMwY2xYNm9LYnJJaWU1T1hmTktJTjRMeTR2N1FmZng2VnlzQW1nUlJWMmpzQkQxM1JqaUR1a0hScWFQallCUlU9IHJvb3RAY2xpZW50LnRlc3Q=']}

print("PUBLIC KEY::", client.host.conn.run("cat /home/user1/.ssh/id_rsa.pub").stdout)
PUBLIC KEY:: ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQC+lu0fXgwXGuVcOz41bOKPtn8Omn05eCmTMr3C6C9CYQWYQYRKlCvTqq5i7ljmcWEFk/dkNQYfL2pWL2q5z9wO2JJHLA1bpDHu2EJOazqucaWWt4nOAdSMFCKb7Dle/kDY84qHN4gFDUGiGXw4zNO3usXGqdr3csx9OkbXI2lACauNm1QMg6Ebpm+vij9YUtf1BOKuw8kGcuShIeerWuth5DIjVk3jAbNmIdZ51NwVgv8QqDzJ5Y/6Hv6kGaVzZrKvYI2LqsGtWDCDTo/2sEXL56YOK9xJVKog/u4/bSgrkoWX4B+acM8mOSBBkLCdOhlelv3b4nOHB8vuAc0lGwuAarDzi1IElDtwP5PY1FuD5hfCzBFul+wrGO8IgWw+Z9aYOXDtEcLkxndszPSncNQAMb+3AgUHWnmw1+xtMz27FQqqjdG9iaLLnfMs+q7sHWNc0clX6oKbrIie5OXfNKIN4Ly4v7Qffx6VysAmgRRV2jsBD13RjiDukHRqaPjYBRU= root@client.test

LS:: total 8.0K
-rw-r--r--. 1 user1 user1  570 Dec 12 23:45 id_rsa.pub
-rw-------. 1 user1 user1 2.6K Dec 12 23:45 id_rsa
```

Checking the generation of the keys
```
    keys = client.tools.sshkey.generate(result.name, result.home)
    print(keys)

PASSED [100%]('ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCo4qTwgYptXapqwg9Hq7/eGqO0ohsZjTXlmCFOQJ8l1kQYuFxJbHkDlIP9rzA5XWVcxi0DaciZQ0H4YpBVg4cprmdW7bhRbr7kZyuAKuGE9ZxhjK1VhC3oHsxT2gm2ez8r+ApbQsOcfE9dNUCtlH0vs37vFfku6/8d8ZhRbXiTJtp0SP3HDcYzisEqasqAOy31NvlxPaRX4V/+AcSJV7+FDIasoePSyt73GDgiuymIddKZMIJopYj9COTTFkzboKet7XzHSe49FdIwQBHHcooqVl2MLz6k5m4dq4SNcAU/IhKSO5iBCy4oOES0H2VXVZPRWVtRzhcAqrGltTuQA0H6Q0B9IyZlIyP7i/eMrd+NrojDRCL5tv2ZJHfhc5x0DY9AsEWzM7psP1YS7FgDWOjKoqSFhVpjkWkpzNS3DMpM/g/4jdQ9H5ydv7lVuZvgQj2oiQoAZCY5uHvgKXimOl1i/rzVZCqFnilFDUNLCys6VDf1q+0Qgv3n2ZUzA0VjQc0= root@client.test', '

----SNIP----
```

In regards to the return code, 0 if the user is found, and doesn't care if the key is found or not, just that the user exists. So for this test to work, we have to assert the output.